src/Kiota.Builder/EqualityComparers/OpenApiServerComparer.cs

```diff
@@ -1,23 +1,38 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
+using System.Text;
 using System.Text.RegularExpressions;
 using Microsoft.OpenApi.Models;
 
 namespace Kiota.Builder.EqualityComparers;
 
-internal sealed partial class OpenApiServerComparer : IEqualityComparer<OpenApiServer>
+internal sealed class OpenApiServerComparer : IEqualityComparer<OpenApiServer>
 {
-    [GeneratedRegex("^https?://", RegexOptions.IgnoreCase, 500)]
-    private static partial Regex protocolCleanupRegex();
+    private static ReadOnlySpan<char> TrimProtocol(ReadOnlySpan<char> output)
+    {
+        output = output.TrimStart("http://");
+        output = output.TrimStart("https://");
+        return output;
+    }
     public bool Equals(OpenApiServer? x, OpenApiServer? y)
     {
-        return x != null && y != null && GetHashCode(x) == GetHashCode(y);
+        if (x is null || y is null) return object.Equals(x, y);
+
+        var x0 = TrimProtocol(x.Url);
+        var y0 = TrimProtocol(y.Url);
+        return MemoryExtensions.Equals(x0, y0, StringComparison.OrdinalIgnoreCase);
```

Comment:
```suggestion
        return x0.Equals(y0, StringComparison.OrdinalIgnoreCase);
```

Reply:
I used the extension because x0 might be null

Reply:
I've updated the null check before to allow using the extension method